source/m2mobject.cpp

```diff
@@ -45,25 +46,14 @@ M2MObject::~M2MObject()
         for (; it!=_instance_list.end(); it++, index++ ) {
             //Free allocated memory for object instances.
             obj = *it;
-
-            char *obj_inst_id = (char*)malloc(BUFFER_SIZE);
-            if(obj_inst_id) {
-                snprintf(obj_inst_id, BUFFER_SIZE,"%d",index);
-
-                String obj_name = M2MBase::name();
-                obj_name += String("/");
-                obj_name += String(obj_inst_id);
-
-                free(obj_inst_id);
-                remove_resource_from_coap(obj_name);
-            }
-
             delete obj;
             obj = NULL;
         }
+        remove_resource_from_coap(M2MBase::name());
```

Comment:
This will not clean the Object Instances under this Object, if Object itself is not part of the registration message. We still need to keep the ObjectInstance removal logic in place.